### ISSUE TITLE
Variable in test name

### ISSUE THREAD

**Issue by mrjcap** [pester/Pester#1958]
## General summary of the issue

When running a Pester test against a remote computer, if the test name contains a variable, Pester will not return the value of the variable. 

## Describe your environment
Pester version     : 5.2.1 C:\Program Files\WindowsPowerShell\Modules\Pester\5.2.1\Pester.psm1
PowerShell version : 5.1.14409.1018
OS version         : Microsoft Windows NT 6.1.7601 Service Pack 1

## Steps to reproduce

I have created a repo with an example here: https://github.com/mrjcap/WinReleaseInfo

```powershell
$invokeCommandSplat = @{
    ScriptBlock    = {
        Import-Module "\\servername\Validation\Pester\5.2.1\Pester.psm1" -Verbose
        $invokePesterSplat = @{
            Path      = "\\servername\Validation\Windows.tests.ps1"
            TagFilter = "Settings"
            PassThru  = $true
            Verbose   = $true
            Output    = 'Detailed'
        }
        Invoke-Pester @invokePesterSplat
    }
    ComputerName   = 'PC01'
    Authentication = 'Credssp'
    Credential     = (Get-Secret -Name domain)
}

Invoke-Command @invokeCommandSplat
```

## Expected Behavior
 Describing Windows
    Context Windows Updates,Build Number and Update Build Revision
VERBOSE: Inside It. Build is 19043
      [+] Windows Build Number should be the latest: **19043** 118ms
VERBOSE: Inside It. Build is 985
      [+] Windows Update Build Revision should be the latest: **985** 6ms
Tests completed in 1.25s
Tests Passed: 2, Failed: 0, Skipped: 0, Pending: 0, Inconclusive: 0 

## Current Behavior
Describing Windows
 Context Windows Updates,Build Number and Update Build Revision
VERBOSE: Inside It. Build is 19043
   [+] Windows Build Number should be the latest:  245ms (184ms|61ms)
VERBOSE: Inside It. Build is 985
   [+] Windows Update Build Revision should be the latest:  9ms (8ms|1ms)
Tests completed in 1.7s
Tests Passed: 2, Failed: 0, Skipped: 0 NotRun: 0

However, I am getting different results in Pester 4.10.1 see the output below.

## Behavior in Pester 4.10.1

```powershell
$invokeCommandSplat = @{
    ScriptBlock    = {
        Import-Module '\\servername\Validation\Pester\4.10.1\Pester.psm1' -Verbose
        $invokePesterSplat = @{
            Path       = '\\servername\Validation\Validation\Windows.tests.ps1'
            Tag        = 'Settings'
            PassThru   = $true
            Verbose    = $true
            OutputFile = '\\servername\Validation\Test-Result.xml'
        }
        Invoke-Pester @invokePesterSplat
    }
    ComputerName   = 'PC01'
    Authentication = 'Credssp'
    Credential     = (Get-Secret -Name domain)
}

Invoke-Command @invokeCommandSplat
```
## Output

    Context Windows Updates,Build Number and Update Build Revision
VERBOSE: Inside It. Build is 19043
      [+] Windows Build Number should be the latest: **19043** 162ms
VERBOSE: Inside It. Build is 985
      [+] Windows Update Build Revision should be the latest: **985** 7ms
Tests completed in 1.27s
Tests Passed: 2, Failed: 0, Skipped: 0, Pending: 0, Inconclusive: 0

**Comment by fflaten** [comment]
Thanks for the report. 

This doesn't have anything to do with remoting, it just makes the issue very visible because you create a new powershell-session (process) on the remote computer every time you use `Invoke-Command` (because you don't save and reuse the session). If you run your tests in a brand new powershell-session locally you'd get the same issue the first time. It would work in the second run because your powershell session is now "dirty" since you now have new global-variables that were missing the first time.

There's a fundamental difference between Pester v4 and Pester v5. Pester v5 runs your testfile twice:
1. Discover-phase: Find and filter the tests.
   - Your testname-string is also expanded here for normal variables. At this point, the global variable doesn't exist yet when executed in a clean powershell-session.
2. Run-phase: Executed the filtered tests. 
   -  The code in `BeforeAll` and `It`-blocks will be executed here.

In Pester v4 all of this were done in a single run, which means the global variables were available when the testname-string was expanded the first time by PowerShell.

**TLDR:** You're using a variable in the test-name that isn't defined yet.
**Solution:** Pester provides a variable-syntax of `<varname>` that you can use in you testname. They won't be expanded until the Run-phase after `BeforeAll` has been executed. Ex:

```powershell
It "Windows Build Number should be the latest: <Major>" {
```

**Tip:** avoid global variables in scripts whenever possible. They persist in the session, so they kinda corrupt your next run.

**Comment by mrjcap** [comment]
@fflaten,

You're absolutely right about the global variables. I was just using them because I thought there was some weird scoping issue.

Thank you for the detailed response!
